Geometry/HcalCommonData/interface/CaloSimParametersFromDD.h

```diff
@@ -2,24 +2,34 @@
 #define HcalCommonData_CaloSimParametersFromDD_h
 
 #include "DetectorDescription/Core/interface/DDsvalues.h"
+#include "DetectorDescription/Core/interface/DDCompactView.h"
+#include <DetectorDescription/DDCMS/interface/DDCompactView.h>
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include <string>
+#include <vector>
 
-class DDCompactView;
 class DDFilteredView;
 class CaloSimulationParameters;
 
 class CaloSimParametersFromDD {
 public:
-  CaloSimParametersFromDD(bool fromDD4Hep);
+  CaloSimParametersFromDD() {}
   virtual ~CaloSimParametersFromDD() {}
 
   bool build(const DDCompactView*, CaloSimulationParameters&);
+  bool build(const cms::DDCompactView*, CaloSimulationParameters&);
 
 private:
+  bool buildParameters(const CaloSimulationParameters&);
   std::vector<std::string> getNames(const std::string&, const DDsvalues_type&, bool);
   std::vector<int> getNumbers(const std::string&, const DDsvalues_type&, bool);
 
-  bool fromDD4Hep_;
+  template <typename T>
```

Comment:
Perhaps, a private utility function can be moved out of the class API? Also, it does not have to be templated:
```
void myPrint(const std::string& value, auto& C) {
  for( auto e : C ) {
   ...
  }
}
```

Reply:
It is moved out of class API but "auto" gives compilation warning. So I have kept the template for the time being